src/reducers/ast.js

```diff
@@ -167,4 +179,8 @@ export function getPreview(state: OuterState) {
   return state.ast.get("preview");
 }
 
+export function getMetaData(state: OuterState) {
```

Comment:
hmm, i think this should be `getSourceMetaData(state: OuterState, source: Source)` so that we scope it down.

Reply:
What do we need source for in this selector?